DEHPEcosimPro/ViewModel/Rows/ParameterDifferenceViewModel.cs

```diff
@@ -0,0 +1,324 @@
+            {
+                OldValue = setOfOldValues.ToString();
+            }
+            else
+            {
+                OldValue = setOfOldValues.FirstOrDefault();
+            }
+
+
+            if (isOptions && isState)
+            {
+                Name = Name + $"\\{this.listofsetOfNewValues[index].ActualOption.ShortName}\\{this.listofsetOfNewValues[index].ActualState.ShortName}";
+            }
+            else if (isOptions && !isState)
+            {
+                Name = Name + $"\\{this.listofsetOfNewValues[index].ActualOption.ShortName}";
+            }
+            else if (!isOptions && isState)
+            {
+                Name = Name + $"{this.listofsetOfNewValues[index].ActualState.ShortName}";
+            }
+            else if (!isOptions && !isState)
+            {
+                Name = this.ModelCode();
+            }
+
+            this.CalculateDiff(OldValue, NewValue, out string Difference, out string PercentDiff);
+
+            return new ParameterDifferenceRowViewModel(this.OldThing, this.NewThing, Name, OldValue, NewValue, Difference, PercentDiff);
+
+        }
+
+        /// <summary>
+        /// Add The valueSet to list
+        /// </summary>
+        /// <param name="option"><see cref="Option"/></param>
+        /// <param name="state"><see cref="ActualFiniteState"/></param>
+        private void PopulateListOfSets(Option option, ActualFiniteState state)
+        {
+            try
+            {
+                this.listofsetOfOldValues.Add(this.OldThing.QueryParameterBaseValueSet(option, state));
+                this.listofsetOfNewValues.Add(this.NewThing.QueryParameterBaseValueSet(option, state));
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Calculate the difference between the old and new value, if possible
+        /// </summary>
+        /// <param name="OldValue"></param>
+        /// <param name="NewValue"></param>
+        /// <param name="Difference">a number, positive or negative (with + or - sign)</param>
+        /// <param name="PercentDiff">a number in percent, positive or negative (with + or - sign)</param>
+        private void CalculateDiff(object OldValue, object NewValue, out string Difference, out string PercentDiff)
+        {
+            Difference = "0";
+            PercentDiff = "0";
+
+            NumberStyles style = NumberStyles.Number | NumberStyles.AllowDecimalPoint;
+            CultureInfo culture = CultureInfo.InvariantCulture;
+
+            var isOldValueDecimal = decimal.TryParse(OldValue.ToString(), style, culture, out decimal decimalOldValue);
+            var isNewValueDecimal = decimal.TryParse(NewValue.ToString(), style, culture, out decimal decimalNewValue);
+
+            if (isOldValueDecimal && isNewValueDecimal)
+            {
+                var diff = decimalNewValue - decimalOldValue;
```

Comment:
This is a bit convoluted?

```
var diff = newValue - oldValue;
Difference = diff.ToString();
PercentDiff = Math.Round((abs(diff)/abs(oldValue))*100, 2);
```
should do the same thing no?

Reply:
i would like to keep the signs + and - before the values
but yes, it could be simplified

Reply:
Ah, indeed I did not think about loosing the + in case of a positive. But then still you can one line it:

```
Difference = diff > 0 ? $"+{diff}" : $"{diff}";
```